lib/publiccloud/instance.pm

```diff
@@ -326,18 +326,23 @@ sub wait_for_guestregister {
         if ($out =~ m/inactive$/) {
             diag("guestregister inactive");
             $self->upload_log($log, log_name => $name);
-            return time() - $start_time;
+            return 1;
         }
         elsif ($out =~ m/failed$/) {
             diag("guestregister failed");
-            $out = $self->run_ssh_command(cmd => 'sudo systemctl status guestregister', quiet => 1);
-            $self->upload_log($log, log_name => $name);
-            return time() - $start_time;
+            # we have some cases where it is known that guestregister service will fail
+            # ( e.g. when we testing images not published on Market hence w/o product codes)
+            return 1 if (get_var('PUBLIC_CLOUD_IGNORE_GUESTREGISTER_FAILURE'));
+            die('guestregister failed');
         }
         elsif ($out =~ m/active$/) {
             diag("guestregister active");
-            $self->upload_log($log, log_name => $name);
-            die "guestregister should not be active on BYOS" if (is_byos);
+            if (is_byos) {
+                die "guestregister should not be active on BYOS";
+            }
+            else {
```

Comment:
You can drop this `else` block.

Reply:
Why drop? Here logs uploaded.

Reply:
> Why drop? Here logs uploaded. AH you mean plain command, out of `if`.

Y mean what's inside the `else` can be taken out and avoid indentation given that we `die` in the `if`.